packages/medusa/src/interfaces/csv-parser.ts

```diff
@@ -0,0 +1,34 @@
+import { AwilixContainer } from "awilix"
+import { IParser } from "./abstract-parser"
+
+export type CsvParserContext = {
+  line: number
+  column: string
+}
+
+export interface IValidator<TLine> {
+  validate: (
+    value: string,
+    line: TLine,
+    context: CsvParserContext
+  ) => Promise<boolean>
+}
+
+export abstract class AbstractValidator<TLine> implements IValidator<TLine> {
+  constructor(protected readonly container: AwilixContainer) {}
+
+  abstract validate(value: string, line: TLine, context: CsvParserContext)
+}
+
+export type CsvSchema<TLine = any> = {
+  columns: {
+    name: string
+    required?: boolean
+    mapTo?: string
+    validator?: AbstractValidator<TLine>
+  }[]
+}
+
+export interface ICsvParser<TResult> extends IParser<TResult> {
```

Comment:
suggestion: this interface can be removed and the delimiter become a protected reaonly property of the class only assign in the constructor